File: src/controllers/authController.js
const userService = require('../services/userService');

/**
 * POST /api/auth/login
 */
const login = (req, res, next) => {
  try {
    const { email, password } = req.body;

    if (!email || !password) {
      return res.status(400).json({
        success: false,
        error: 'Email and password are required',
      });
    }

    const user = userService.verifyPassword(email, password);

    if (!user) {
      return res.status(401).json({
        success: false,
        error: 'Invalid email or password',
      });
    }

    // Generate a mock JWT for testing purposes
    const mockToken = `mock_jwt_token_for_${user.id}_${Buffer.from(user.email).toString('base64')}`;

    res.status(200).json({
      success: true,
      message: 'Login successful',
      token: mockToken,
      user,
    });
  } catch (err) {
    next(err);
  }
};

module.exports = { login };

File: src/main/java/com/LeNhatTruong/automation/controller/AuthController.java
package com.LeNhatTruong.automation.controller;

import com.LeNhatTruong.automation.model.User;
import com.LeNhatTruong.automation.service.UserService;
import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.*;

import java.util.Base64;
import java.util.HashMap;
import java.util.Map;

@RestController
@RequestMapping("/api/auth")
public class AuthController {

    private final UserService userService;

    public AuthController(UserService userService) {
        this.userService = userService;
    }

    @PostMapping("/login")
    public ResponseEntity<Map<String, Object>> login(@RequestBody Map<String, String> request) {
        String email = request.get("email");
        String password = request.get("password");

        if (email == null || password == null) {
            Map<String, Object> response = new HashMap<>();
            response.put("success", false);
            response.put("error", "Email and password are required");
            return new ResponseEntity<>(response, HttpStatus.BAD_REQUEST);
        }

        User user = userService.verifyPassword(email, password);

        if (user == null) {
            Map<String, Object> response = new HashMap<>();
            response.put("success", false);
            response.put("error", "Invalid email or password");
            return new ResponseEntity<>(response, HttpStatus.UNAUTHORIZED);
        }

        String mockToken = "mock_jwt_token_for_" + user.getId() + "_" + Base64.getEncoder().encodeToString(user.getEmail().getBytes());

        Map<String, Object> response = new HashMap<>();
        response.put("success", true);
        response.put("message", "Login successful");
        response.put("token", mockToken);
        response.put("user", user);

        return new ResponseEntity<>(response, HttpStatus.OK);
    }
}
